presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfoResource.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.execution.resourceGroups.ResourceGroupManager;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import io.airlift.json.JsonCodec;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+@Path("/v1/resourceGroupState")
+public class ResourceGroupStateInfoResource
+{
+    private final ResourceGroupManager resourceGroupManager;
+    private final JsonCodec<ResourceGroupId> resourceGroupIdJsonCodec = JsonCodec.jsonCodec(ResourceGroupId.class);
```

Comment:
There are a lot of places that are neither static nor upper case. Which ones are you referring to?